### PR TITLE
Add support for tables with AggregateFunction columns

### PR DESCRIPTION
**Pull request by mjc-gh**

### CODE REVIEW

File: lib/clickhouse-activerecord/schema_dumper.rb [hunk 1]
```diff
@@ -147,8 +147,11 @@ def format_colspec(colspec)
     end
 
     def schema_limit(column)
-      return nil if column.type == :float
-      super
+      if column.type == :float
+        return 8 if column.sql_type == "Float64"
```
Comment (danielwestendorf) [inline lib/clickhouse-activerecord/schema_dumper.rb:152]: Looks like CH supports `Float32` as well.

```suggestion
        return 4 if column.sql_type == "Float32"
        return 8 if column.sql_type == "Float64"
```

File: lib/clickhouse-activerecord/schema_dumper.rb [hunk 2]
```diff
@@ -172,6 +175,16 @@ def schema_low_cardinality(column)
       (column.sql_type =~ /LowCardinality\(/).nil? ? nil : true
     end
 
+    def schema_aggregate_function(column)
+      match = column.sql_type.match(/AggregateFunction\((.+), (\w+)\)/)
+
+      return {} if match.nil? || match.size != 3
+
+      { aggregate_function: match[1].inspect }.tap do |spec|
+        spec[:limit] = 8 if match[2] == "Float64"
```
Comment (danielwestendorf) [inline lib/clickhouse-activerecord/schema_dumper.rb:186]:
```suggestion
        spec[:limit] = 4 if match[2] == "Float32"
        spec[:limit] = 8 if match[2] == "Float64"
```

File: lib/active_record/connection_adapters/clickhouse/table_definition.rb [hunk 3]
```diff
@@ -63,6 +63,13 @@ def integer(*args, **options)
           args.each { |name| column(name, kind, **options.except(:limit, :unsigned)) }
         end
 
+        def float(*args, **options)
+          kind = :float
+          kind = :float64 if options[:limit] == 8
```
Comment (danielwestendorf) [inline lib/active_record/connection_adapters/clickhouse/table_definition.rb:69]:
```suggestion
          kind = :float32 if options[:limit] == 4
          kind = :float64 if options[:limit] == 8
```

**Review by danielwestendorf — APPROVED** [review]
Let's merge this after the recent gem upgrade bakes for a day. We can also upstream.

File: lib/clickhouse-activerecord/schema_dumper.rb (hunk 1 above)
Comment (mjc-gh) [reply lib/clickhouse-activerecord/schema_dumper.rb:152]: Yeah the implied default for `float` is `Float32` currently:

https://github.com/huntresslabs/clickhouse-activerecord/blob/337231e4ba443f8dedbea4e8383e23dd06d601ae/lib/active_record/connection_adapters/clickhouse_adapter.rb#L89

**Pull request merged**